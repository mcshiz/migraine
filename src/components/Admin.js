import React from 'react'
import '../styles/admin.css'
class Admin extends React.Component {
    constructor() {
        super();
        this.clearData = this.clearData.bind(this);
        this.loadExampleData = this.loadExampleData.bind(this)

    }
    clearData(){
        localStorage.clear();
        alert('Data Cleared');
    }

    loadExampleData() {
        let exampleData = [
            {date: "2017-12-23T17:33:33.265Z", obj: {"pain":"3","work":"9","sleep":"8","notes":"notes","pressure":1025}},
            {date: "2017-12-24T17:33:33.265Z", obj: {"pain":"4","work":"2","sleep":"12","notes":"notes","pressure":1029}},
            {date: "2017-12-25T17:33:33.265Z", obj: {"pain":"6","work":"5","sleep":"8","notes":"notes","pressure":1030}},
            {date: "2017-12-26T17:33:33.265Z", obj: {"pain":"9","work":"3","sleep":"6","notes":"notes","pressure":1028}},
            {date: "2017-12-27T17:33:33.265Z", obj: {"pain":"6","work":"0","sleep":"4","notes":"notes","pressure":1026}},
            {date: "2017-12-28T17:33:33.265Z", obj: {"pain":"3","work":"1","sleep":"9","notes":"notes","pressure":1026}},
            {date: "2017-12-29T17:33:33.265Z", obj: {"pain":"1","work":"3","sleep":"10","notes":"notes","pressure":1020}},
        ];
        exampleData.map((obj) => {
            localStorage.setItem(obj.date, JSON.stringify(obj.obj))
        });
        alert("Data Loaded")

    }
    render() {
        return (
            <div>
                <div>
                    <button className="button warning clear" onClick={this.clearData}>Clear Data</button>
                </div>
                <div>
                    <button className="button warning load-example-data" onClick={this.loadExampleData}>Load Example Data</button>
                </div>
            </div>
        )
    }
}

export default Admin