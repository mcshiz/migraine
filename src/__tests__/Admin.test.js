import React from 'react'
import Admin from '../components/Admin'

// Not Testing Clear LocalStorage or Load Example Data
// that should be covered by localStorage itself (duplicate testing)
describe('Admin Component', () => {
    const wrapper = shallow(
        <Admin/>
    );
    it('Should compare the component with a snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('Should clear localStorage', () => {
       wrapper.find('.button.warning.clear').simulate('click');
        expect(localStorage.clear).toHaveBeenCalledTimes(1);
        expect(localStorage.length).toBe(0);
    });
});
