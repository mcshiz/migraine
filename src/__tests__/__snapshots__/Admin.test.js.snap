// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Component Should compare the component with a snapshot 1`] = `
<div>
  <div>
    <button
      className="button warning clear"
      onClick={[Function]}
    >
      Clear Data
    </button>
  </div>
  <div>
    <button
      className="button warning load-example-data"
      onClick={[Function]}
    >
      Load Example Data
    </button>
  </div>
</div>
`;
